homeassistant/components/kef/media_player.py

```diff
@@ -190,6 +281,9 @@ def state(self):
                 state = await self._speaker.get_state()
                 self._source = state.source
                 self._state = STATE_ON if state.is_on else STATE_OFF
+                if self._dsp is None:
```

Comment:
I am wondering if you need to drop the update dsp service and just do this once every minute or so?

Reply:
That way we don't require each user to configure this themselves.

Reply:
I am not sure about it.

The function call creates a lot of calls to the speaker and overloading it makes it crash sometimes.

Reply:
Maybe update it once every 30 minutes then, and maybe have some delays between the requests ? It feels weird to provide information but expect users to hit an update button.

Reply:
Yes, sounds good to do it infrequently.

What's the best way of adding this?

Also, I've added https://github.com/home-assistant/core/pull/31967/commits/0f9bc20d30c142b9c9c0fb059e8e3a8d9c6c4050, such that `update_dsp` is only called when a setting has been updated.

Reply:
Is this (https://github.com/home-assistant/core/pull/31967/commits/8d0dd924d2a5b563d2f7b35231dad046e1f5cfb3) the right way to schedule the updates?

Reply:
@balloob, you replied in the commits. I will continue here.

I've implemented all your suggestions.